internal/merger/merger.go

```diff
@@ -213,43 +214,54 @@ func MergeEvent(main, aux *pbv2.EventResponse) *pbv2.EventResponse {
 	return main
 }
 
-// MergeObservation merges two V2 observation responses.
-func MergeObservation(main, aux *pbv2.ObservationResponse) *pbv2.ObservationResponse {
+// MergeObservation merges multiple V2 observation responses.
+func MergeObservation(main *pbv2.ObservationResponse, aux []*pbv2.ObservationResponse) *pbv2.ObservationResponse {
```

Comment:
I wonder if using https://gobyexample.com/variadic-functions is cleaner? And saying the args are ranked by preference?

Reply:
Good idea, thanks! Added a wrapper recursive method